app/models/post.rb

```diff
@@ -1,8 +1,28 @@
 class Post < ApplicationRecord
   include Elasticsearch::Model
+  include PgSearch::Model
 
   validates :title, :body, :topic, presence: true
 
+  # Full-text search
+  # multisearchable against: [:title, :body, :topic]
+  pg_search_scope(
+    :search_full_text,
+    against: {
+      body: 'A',
+      title: 'B',
+      topic: 'C'
+    },
+    using: {
+      tsearch: {
+        dictionary: 'english',
+        any_word: true,
+        prefix: true
+      },
+    }
+  )
+  
+  # ElastichSearch
   alias_method :to_hash, :attributes
 
   settings number_of_shards: 1 do
```

Comment:
line 32 below: `indexes :topic, type: :keyword`

what's the difference of `:keyword` and `:text` ? do we have the same in fulltext?

Reply:
According to the [Elastic docs](https://www.elastic.co/guide/en/elasticsearch/reference/current/keyword.html):

> Keyword is used for structured content such as IDs, email addresses, hostnames, status codes, zip codes, or tags.

> Avoid using keyword fields for full-text search. Use the [text](https://www.elastic.co/guide/en/elasticsearch/reference/current/text.html) field type instead.

The main difference is that `text` data type is analysed at the time of indexing, and `keyword` type is not.

In terms of the comparison with fulltext, I don't think there is a similar concept. But you could choose to store the indexed values or not, so it's still possible to get the same behaviour is some way